### PR TITLE
[TechInsights] log errors when tech insights throws an error

### PR DESCRIPTION
**Pull request by goenning**
Signed-off-by: goenning <me@goenning.net>

## Hey, I just made a Pull Request!

We spent a couple of hours trying to understand why our handler was not working in certain scenarios, until we found that when the `handler` of a fact retriever throws an error, it doesn't get logged anywhere and the job just exit silently.

Developers are still encouraged to handle the errors themselves in the handler, but in case they forget and purposely do not add them, this PR will at least ensure we have a trace of what happened. No facts are stored when an error is thrown.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.ts [hunk 1]
```diff
@@ -108,15 +109,24 @@ export class FactRetrieverEngine {
...
-            factRetriever.id
-          } in ${duration(startTimestamp)}`,
+
+      let facts: TechInsightFact[] = [];
+      try {
+        facts = await factRetriever.handler({
+          ...this.factRetrieverContext,
+          entityFilter: factRetriever.entityFilter,
+        });
+        if (this.logger.isDebugEnabled()) {
```
Comment (benjdlambert) [inline plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.ts:119]: Think we can remove this line as `this.logger.debug` will only log things if the log level is `debug`

**Review by benjdlambert — COMMENTED** [review]
Thanks for this! Just a small nit, and also @Xantier do you want to take a quick peek?

File: plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.ts (hunk 1 above)
Comment (Xantier) [reply plugins/tech-insights-backend/src/service/fact/FactRetrieverEngine.ts:119]: I think it's my JVM past leaking through. Micro-optimization to not do an expensive string interpolation if it's not needed. Can indeed be removed.

**Review by Xantier — APPROVED** [review]
Yup, this looks good to me. Nitpick optional

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Sweet - think we're ready to 🚢  when ✅

**Comment by goenning** [comment]
@benjdlambert can we :shipit: ?

**Comment by benjdlambert** [comment]
Apologies @goenning just seen this message come through in the notifications - see that it's been taken care of 👍

**Pull request merged**